### ISSUE TITLE
UI: Allow hiding or collapsing of less used channels

### ISSUE THREAD

**Issue by ssbarnea** [thelounge/thelounge#4078]
### Feature Description

I am heavy IRC user and I am joined in >50 channels related to open-source and the channel left sidebar no longer fits a full screen, making quite hard to spot the channels that matters from those where I am only lurking (or waiting for someone to contact me, to to mention a keyword).

I am not yet sure which is the easiest solution for addressing this issue so I will list few ideas, letting other review them, hopefully we can find a solution for **scaling UI beyond ~15 channels**:

- **grouping channels** : be able to right click on a channel and select a pick a group. A group would be listed at root level with `[+]` icon and should show the total number of messages it contains. Selecting the group should expand it, clicking again on it should collapse it.
- **colors** : ability to change the color used to display channels, this would make easier to visually group channels. Color should be selectable using right click on channel, having a preset of ~6-10 colors should be more than enough.
- **auto-hide channel** : ability to hide channels without any activity in them. This can go at two levels: hide if there are no messages and hide if there are no mentions.

**Comment by xPaw** [comment]
> auto-hide channel  [..] hide if there are no mentions.

This is already how collapsing works, click the arrow button next to the network name.

**Comment by ssbarnea** [comment]
> This is already how collapsing works, click the arrow button next to the network name.

Yes, but we need collapse for groups of channels, so we de-clutter the UI.

**Comment by venatiodecorus** [comment]
I like the way IRC Cloud handles it, where you can pin your favorite channels to the top of the list. I'd love to see that feature.

**Comment by ssbarnea** [comment]
@xPaw Is there a way to hide specific channels? The option to collapse a network does not do any good if you are active on some channels on that network but not on others. Probably I could use two connections with separated nicks for freenode, one for my active channels and one for occasional ones, ones that I want to be joined into but where I do not expect to chat very often, and keep the second one collapsed. It would be a PITA and confuse people that discover I am using two nicks.

**Comment by brunnre8** [comment]
@ssbarnea please don't ping random maintainers, it's rather rude.

Second, that's a support question and not specific to this bug per se, please use our IRC chan on freenode.

Hop in there and say hi (I'm bookworm)

You can do what you want with a single nick by using ZNC as long as TL doesn't natively support this.
Again, hop into IRC and I can explain the details

**Comment by brettgilio** [comment]
It is really not the most elaborate work around, but in the mean time

```
div[data-name="#channel-name"] {
    display: none;
}
```
